src/librustc_resolve/lib.rs

```diff
@@ -3181,61 +3191,53 @@ impl<'a> Resolver<'a> {
     // Checks that all of the arms in an or-pattern have exactly the
     // same set of bindings, with the same binding modes for each.
     fn check_consistent_bindings(&mut self, pats: &[P<Pat>]) {
-        if pats.is_empty() {
+        if pats.len() <= 1 {
             return;
```

Comment:
Shouldn't this be `bug!("An or-pattern cannot have < 2 patterns by construction");`?

Reply:
@Centril, not really, all arms of a `match` go through this check, including those with a single pattern. 🙂

Reply:
Heh... can we float that check to those locations instead? As-is that doc comment is pretty misleading so at least if the code stays the same the comment should be changed.